### PR TITLE
Implement table record virtualizer back

### LINKED ISSUE #2728: Implement table record virtualizer back

### PR DESCRIPTION
**Pull request by gitstart-twenty**
Fixes #2728

### CODE REVIEW

File: front/src/modules/ui/object/record-table/record-table-cell/components/RecordTableCellContainer.tsx [hunk 1]
```diff
@@ -114,6 +116,7 @@ export const TableCellContainer = ({
       <StyledCellBaseContainer
         onMouseEnter={handleContainerMouseEnter}
         onMouseLeave={handleContainerMouseLeave}
+        style={{ width }}
```
Comment (gitstart-twenty) [inline front/src/modules/ui/object/record-table/record-table-cell/components/RecordTableCellContainer.tsx:119]: Hey @charlesBochet,

The virtualizer's requirement for absolute positioning is affecting the table body layout. As a result, the natural dimensions are compromised, and this impacts the cell widths. To address this, I've manually adjusted the widths to match those of the table header cells.

File: front/package.json [hunk 2]
```diff
@@ -16,6 +16,7 @@
     "@hookform/resolvers": "^3.1.1",
     "@sniptt/guards": "^0.2.0",
     "@tabler/icons-react": "^2.30.0",
+    "@tanstack/react-virtual": "^3.0.0",
```
Comment (charlesBochet) [inline front/package.json:19]: @gitstart-twenty let's not re-add tanstack, I don't think it's useful actually, it's overkill and the API of this library is not great.

I think we need something simpler.
Could you create a component Virtualizer in ui/utilities/virtualizer taking the following code: https://github.com/NightCafeStudio/react-render-if-visible/blob/main/src/render-if-visible.tsx (it seems to be what we need but not maintained anymore)

Then use it in RecordTableBody to wrap a row. Should be simpler and achieving the same result.
I don't think we need to virtualize commands for now btw :)

**Review by charlesBochet — COMMENTED** [review]
@gitstart-twenty I've left comment and proposed another strategy

File: front/package.json (hunk 2 above)
Comment (gitstart-twenty) [reply front/package.json:19]: Alright, on this

**Review by charlesBochet — APPROVED** [review]
Thank you! Let's merge it and iterate from there :)

**Pull request merged**